streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java

```diff
@@ -350,6 +350,33 @@ public static void waitForCompletion(final KafkaStreams streams,
         return accumData;
     }
 
+    public static <K, V> List<KeyValue<K, V>> waitUntilExactKeyValueRecordsReceived(final Properties consumerConfig,
+                                                                                    final String topic,
+                                                                                    final List<KeyValue<K, V>> expectedRecords) throws InterruptedException {
+        return waitUntilExactKeyValueRecordsReceived(consumerConfig, topic, expectedRecords, DEFAULT_TIMEOUT);
+    }
+
+    public static <K, V> List<KeyValue<K, V>> waitUntilExactKeyValueRecordsReceived(final Properties consumerConfig,
+                                                                                    final String topic,
+                                                                                    final List<KeyValue<K, V>> expectedRecords,
+                                                                                    final long waitTime) throws InterruptedException {
+        final List<KeyValue<K, V>> accumData = new ArrayList<>();
+        try (final Consumer<K, V> consumer = createConsumer(consumerConfig)) {
+            final TestCondition valuesRead = new TestCondition() {
+                @Override
+                public boolean conditionMet() {
+                    final List<KeyValue<K, V>> readData =
+                            readKeyValues(topic, consumer, waitTime, expectedRecords.size());
+                    accumData.addAll(readData);
+                    return accumData.containsAll(expectedRecords);
```

Comment:
This is not a check for "exactly": if `accumData` contains more record than `expectedRecords` this test would pass. Shouldn't we use `equals()` ?

Reply:
Good catch, actually we should check that for each key, the expected record is the last record, I'll try to update the function.